native-image-support/src/main/java/com/google/cloud/nativeimage/features/gcp/CloudSqlFeature.java

```diff
@@ -100,5 +100,34 @@ public void beforeAnalysis(BeforeAnalysisAccess access) {
       // Additional MySQL resources.
       resourcesRegistry.addResourceBundles("com.mysql.cj.LocalizedErrorMessages");
     }
+
+    // Support Unix Domain Socket
+    if (access.findClassByName("jnr.ffi.provider.FFIProvider") != null) {
+      //note: not support AsmLibraryLoader
+      //must execute application with argument "-Djnr.ffi.asm.enabled=false"
+      //try to "bake" the property at build time
+      String asmEnabledPropertyKey = "jnr.ffi.asm.enabled";
+      //if no property set for native-image
+      if (System.getProperty(asmEnabledPropertyKey) == null) {
+        System.setProperty(asmEnabledPropertyKey, String.valueOf(false));
+      }
+      //Provider
+      NativeImageUtils.registerForReflectiveInstantiation(access, "jnr.ffi.provider.jffi.Provider");
+      //bake system property
+      RuntimeClassInitialization.initializeAtBuildTime("jnr.ffi.provider.jffi.NativeLibraryLoader");
+      //StubLoader
+      //StubLoader will extract and load native library, can't init at build time
+      //TODO only cover linux usage yet
+      //since can't use determineOS() and determineCPU() in StubLoader
```

Comment:
How about we replaces this comment with the following? It just summarizes your PR description. 

```suggestion
      // StubLoader loads the native stub library and is only intended to be called reflectively. Note that this configuration only covers linux x86_64 platform at the moment. 
```